server/Connection.cpp

```diff
@@ -33,71 +38,52 @@ Connection& Connection::operator=(Connection const& connection) {
   return *this;
 }
 
-// Public method
+
+/**
+ * Public method - request
+ */
 
 // 요청 읽기
-// - 임시 메서드
+// - 읽기에 실패한 경우 예외 발생
+// - 클라이언트 연결 종료가 감지된 경우 status를 CLOSE로 바꿈
 void Connection::readSocket(void) {
-  _status = ON_RECV;
-  u_int8_t buffer[BUFFER_SIZE];
+  changeStatus(ON_RECV);
 
+  u_int8_t buffer[BUFFER_SIZE];
   memset(buffer, 0, BUFFER_SIZE);
-  ssize_t bytesRead = ::read(_fd, buffer, sizeof(buffer) - 1);
+  ssize_t bytesRead = read(_fd, buffer, sizeof(buffer) - 1);
 
   if (bytesRead < 0) {
-    throw std::runtime_error("read error");
-  } else if (bytesRead == 0) {
-    // 클라이언트가 연결을 종료했음
-    _status = CLOSE;
-    std::cout << "Client: connection closed" << std::endl;
+    throw std::runtime_error("[4000] Connection: readSocket - read fail");
+  } else if (bytesRead == 0) {  // 클라이언트가 연결을 종료했음
+    changeStatus(CLOSE);
+    std::cout << "Client: connection closed" << std::endl;  // debug
+    updateLastCallTime();
     return;
-  } else {
-    try {
-      _requestParser.parse(buffer, bytesRead);
-
-      if (_requestParser.getParsingStatus() == HEADER_FIELD_END) {
-        // Location 블록 찾아오기
-        // Request const& request = _requestParser.getRequest();
-        // std::string path = request.getPath();
-        // std::string host = request.getHeaderFieldValues("Host").front();
-        // Location location("/", "/var/www", "index.html");  // 찾았어
-        // 다시 파싱
-        // _requestParser.setLocation(location);
-        _requestParser.parse(buffer, 0);
-      }
-
-      if (_requestParser.getParsingStatus() == DONE) {
-        Request const& request = _requestParser.getRequest();
-        request.print();
-
-        // 응답 만들기
-        // WRITE 등록
-        _status = TO_SEND;
-
-        Kqueue::removeReadEvent(_fd);
-        Kqueue::addWriteEvent(_fd);
-
-        _requestParser.clear();
-      }
-
-    } catch (std::exception& e) {
-      // TODO: StatusException의 경우 해당하는 에러 코드 전송 및 커넥션 끊기
-
-      _requestParser.clear();
-      throw;
-    }
   }
 
+  parseRequest(buffer, bytesRead);
   updateLastCallTime();
 }
 
+// storage에 있는 요청 읽기
+// - 읽기에 실패한 경우 예외 발생
 void Connection::readStorage(void) {
-  try {
-    _status = ON_RECV;
+  changeStatus(ON_RECV);
 
-    u_int8_t tmp[1];
-    _requestParser.parse(tmp, 0);
+  u_int8_t tmp[1];
+
+  parseRequest(tmp, 0);
+  updateLastCallTime();
+}
+
+// RequestParser에서 요청 읽기
+// - bytesRead를 0으로 하면 storage에 남아있는 내용을 파싱
+void Connection::parseRequest(u_int8_t* buffer, ssize_t bytesRead) {
```

Comment:
여기서 buffer를 `const*`로 보내도 좋을 것 같아요.
```suggestion
void Connection::parseRequest(u_int8_t const* buffer, ssize_t bytesRead) {
```